README


This archive contains the distribution Sub-Name,
version 0.12:

  (re)name a sub

This software is copyright (c) 2004 by Matthijs van Duin and cPanel Inc..

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Sub-Name

0.12      2014-09-04 07:28:59Z
    - fix licence in LICENSE and pod

0.11      2014-08-20 04:44:25Z (TRIAL RELEASE)
    - move variable declaration to fix warning under
      -Werror=declaration-after-statement, to allow compilation under MSVC
      (J.R.Mash, github #3)

0.10      2014-08-16 00:46:35Z (TRIAL RELEASE)
    - converted distribution packaging to Dist::Zilla

0.09      2014-08-09
    - Copy the contents of the %DB::sub entry if it exists; fixes
      Devel::NYTProf's anon sub handling (RT#50524, ilmari)

0.08      2014-08-04
    - Fix leak when setting a fully-qualified name (ilmari, github#1)

0.07      2014-07-11
    - skip optional test if B::C 1.48 is not installed (Reini Urban)

0.06      2014-07-10
    - Do not change the string arg in XS, use copy instead. Fixes perlcc -O3
      RT#96893 (Reini Urban)
    - add README make target
    - add more distribution metadata

0.05      2010-09-07 22:51:00Z
    - Stop using the padlist to refcount GVs. Instead use regular magic. This
      allows various modules, including B::Deparse, to safely peek into pads of
      (re)named subs (Closes RT#42725) (Goro Fuji).
    - Support perl >= 5.13.3 by using the new CvGV_set interface there
      (Closes RT#59558).
    - Stop using the deprecated PL_no_symref (Closes RT#57843).

0.04      2008-07-18 13:23:00Z
    - Fixed for perl 5.6 and 5.005 threads (tested)

0.03      2008-02-20 19:19:00Z
    - Fixed crash when trying to rename xsubs
    - As a side-effect, should work with 5.005 threads (untested)

0.02      2004-08-18 17:51:00Z
    - Fixed documentation, which erroneously mentioned the existence of
      two exported functions.


lib/Sub/Name.pm
package Sub::Name;
# git description: v0.10-TRIAL-4-gf093e04
$Sub::Name::VERSION = '0.11'; # TRIAL
# ABSTRACT: (re)name a sub

#pod =pod
#pod
#pod =head1 SYNOPSIS
#pod
#pod     use Sub::Name;
#pod
#pod     subname $name, $subref;
#pod
#pod     $subref = subname foo => sub { ... };
#pod
#pod =head1 DESCRIPTION
#pod
#pod This module has only one function, which is also exported by default:
#pod
#pod =for stopwords subname
#pod
#pod =head2 subname NAME, CODEREF
#pod
#pod Assigns a new name to referenced sub.  If package specification is omitted in
#pod the name, then the current package is used.  The return value is the sub.
#pod
#pod The name is only used for informative routines (caller, Carp, etc).  You won't
#pod be able to actually invoke the sub by the given name.  To allow that, you need
#pod to do glob-assignment yourself.
#pod
#pod Note that for anonymous closures (subs that reference lexicals declared outside
#pod the sub itself) you can name each instance of the closure differently, which
#pod can be very useful for debugging.
#pod
#pod =head1 SEE ALSO
#pod
#pod =over 4
#pod
#pod =item *
#pod L<Sub::Identify> - for getting information about subs
#pod
#pod =back
#pod
#pod =head1 COPYRIGHT AND LICENSE
#pod
#pod =for stopwords cPanel
#pod
#pod This software is copyright (c) 2004, 2008 by Matthijs van Duin, all rights reserved;
#pod copyright (c) 2014 cPanel Inc., all rights reserved.
#pod
#pod This program is free software; you can redistribute it and/or modify
#pod it under the same terms as Perl itself.
#pod
#pod =cut

use 5.006;

use strict;
use warnings;

use Exporter 5.57 'import';

our @EXPORT = qw(subname);
our @EXPORT_OK = @EXPORT;

use XSLoader;
XSLoader::load(
    __PACKAGE__,
    exists $Sub::Name::{VERSION}
        ? ${ $Sub::Name::{VERSION} }
        : (),
);

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Sub::Name - (re)name a sub

=head1 VERSION

version 0.11

=head1 SYNOPSIS

    use Sub::Name;

    subname $name, $subref;

    $subref = subname foo => sub { ... };

=head1 DESCRIPTION

This module has only one function, which is also exported by default:

=for stopwords subname

=head2 subname NAME, CODEREF

Assigns a new name to referenced sub.  If package specification is omitted in
the name, then the current package is used.  The return value is the sub.

The name is only used for informative routines (caller, Carp, etc).  You won't
be able to actually invoke the sub by the given name.  To allow that, you need
to do glob-assignment yourself.

Note that for anonymous closures (subs that reference lexicals declared outside
the sub itself) you can name each instance of the closure differently, which
can be very useful for debugging.

=head1 SEE ALSO

=over 4

=item *
L<Sub::Identify> - for getting information about subs

=back

=for stopwords cPanel

This software is copyright (c) 2004, 2008 by Matthijs van Duin, all rights reserved;
copyright (c) 2014 cPanel Inc., all rights reserved.

This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=head1 AUTHOR

Matthijs van Duin <xmath@cpan.org>

=head1 CONTRIBUTORS

=for stopwords Karen Etheridge Florian Ragwitz Matthijs van Duin Reini Urban Dagfinn Ilmari Mannsåker gfx J.R. Mash

=over 4

=item *

Karen Etheridge <ether@cpan.org>

=item *

Florian Ragwitz <rafl@debian.org>

=item *

Matthijs van Duin <xmath-no-spam@nospam.cpan.org>

=item *

Reini Urban <rurban@cpanel.net>

=item *

Dagfinn Ilmari Mannsåker <ilmari@ilmari.org>

=item *

gfx <gfuji@cpan.org>

=item *

J.R. Mash <jmash.code@gmail.com>

=back

=head1 COPYRIGHT AND LICENSE

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "(re)name a sub",
  "AUTHOR" => "Matthijs van Duin <xmath\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Sub-Name",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Sub::Name",
  "PREREQ_PM" => {
    "Exporter" => "5.57",
    "XSLoader" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "B::Deparse" => 0,
    "Devel::CheckBin" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "0.12",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "B::Deparse" => 0,
  "Devel::CheckBin" => 0,
  "Exporter" => "5.57",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Test::More" => "0.88",
  "XSLoader" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



